perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/widgets/items/ListBox.java

```diff
@@ -0,0 +1,101 @@
+package cz.metacentrum.perun.wui.registrar.widgets.items;
+
+import com.google.gwt.event.dom.client.BlurEvent;
+import com.google.gwt.event.dom.client.BlurHandler;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.user.client.ui.HasVerticalAlignment;
+import com.google.gwt.user.client.ui.HorizontalPanel;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.gwt.user.client.ui.VerticalPanel;
+import cz.metacentrum.perun.wui.widgets.boxes.ExtendedTextBox;
+import cz.metacentrum.perun.wui.widgets.PerunButton;
+import cz.metacentrum.perun.wui.model.beans.ApplicationFormItemData;
+import cz.metacentrum.perun.wui.registrar.widgets.PerunForm;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents ListBox form item.
+ *
+ * @author Jakub Hejda <Jakub.Hejda@cesnet.cz>
+ */
+public class ListBox extends WidgetBox {
+
+	List<ExtendedTextBox> inputList;
+
+	public ListBox(PerunForm form, ApplicationFormItemData item, String lang) {
+		super(form, item, lang);
+	}
+
+	@Override
+	public String getValue() {
+		String value = "";
+		for (ExtendedTextBox input : inputList) {
+			value += (input.getValue() == null || input.getValue().isEmpty()) ? "" : input.getValue() + ",";
+		}
+
+		return value;
+	}
+
+	@Override
+	protected Widget initWidget() {
+		inputList = new ArrayList<>();
+		return super.initWidget();
+	}
+
+	@Override
+	protected void setValidationTriggers() {
+		for (ExtendedTextBox input : inputList) {
+			input.addBlurHandler(new BlurHandler() {
+				@Override
+				public void onBlur(BlurEvent event) {
+					validateLocal();
+				}
+			});
+		}
+	}
+
+	@Override
+	protected void setValueImpl(String value) {
+		int counter = 0;
+		for (String val : value.split(",")) {
+			if (counter != 0) {
+				generateItemWithRemoveButton((VerticalPanel) widget);
+			}
+			inputList.get(counter).setValue(val.trim());
+			counter++;
+		}
+	}
+
+	@Override
+	protected PerunButton generateAddButton(VerticalPanel vp) {
+		return new PerunButton(getTranslation().addNewValue(), new ClickHandler() {
+			public void onClick(ClickEvent event) {
+				generateItemWithRemoveButton(vp);
+			}
+		});
+	}
+
+	@Override
+	protected void generateItemWithRemoveButton(VerticalPanel vp) {
+		HorizontalPanel hp = new HorizontalPanel();
+		hp.setVerticalAlignment(HasVerticalAlignment.ALIGN_MIDDLE);
+		hp.setWidth("100%");
+		ExtendedTextBox input = new ExtendedTextBox();
```

Comment:
You should set regex from the form item to the `ExtendedTextBox` here, so that we can validate each text box separately. For static widgets like `TextField` its implemented in `initWidget()`, but for this case it should be here.